include/glaze/json/custom.hpp

```diff
@@ -48,11 +48,22 @@ namespace glz
                         (value.val.*(value.from))();
                      }
                      else if constexpr (std::tuple_size_v<Tuple> == 1) {
-                        std::decay_t<std::tuple_element_t<0, Tuple>> input{};
-                        read<json>::op<Opts>(input, ctx, it, end);
-                        if (bool(ctx.error)) [[unlikely]]
-                           return;
-                        (value.val.*(value.from))(input);
+                        using Input = std::decay_t<std::tuple_element_t<0, Tuple>>;
+#if defined(__GNUC__) && (!defined __clang__)
+                        if constexpr (std::is_base_of_v<std::_Function_base, Input>) {
```

Comment:
Is the goal here just to verify that the provided argument is a std::function?

If so, I believe the use of std::_Function_base should be avoided. I don't think that's standardized, and just so happens to be the name of a type in the internal code of libstdc++. Furthermore, libstdc++ can be used as the standard library with clang, so the pre-processor check doesn't target this "feature", or non feature in this case.

Please, correct me if I'm wrong